src/tests/functional/shared_test_classes/src/single_op/tensor_iterator.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/control_flow/unroll_tensor_iterator.hpp"
+#include "shared_test_classes/single_op/tensor_iterator.hpp"
+#include "openvino/pass/manager.hpp"
+#include "ngraph_functions/builders.hpp"
+
+namespace ov {
+namespace test {
+std::string TensorIteratorTest::getTestCaseName(const testing::TestParamInfo<TensorIteratorParams> &obj) {
+    bool should_decompose;
+    size_t seq_lengths;
+    size_t batch;
+    size_t hidden_size;
+    size_t input_size = 10;
+    size_t sequence_axis;
+    ngraph::helpers::TensorIteratorBody ti_body;
```

Comment:
Let's use ov::test::utils... If I remember right all enums from ngraph helpers were moved to new API

Reply:
Unfortunately not all enums are ported, so I ported `TensorIteratorBody` enum